github_pr_to_internal_pr/github_pr_to_internal_pr.py

```diff
@@ -0,0 +1,237 @@
+    GITHUB_TOKEN = os.environ['GITHUB_TOKEN']
+
+    # NOTE: General form is: https://api.github.com/repos/org/repo/collaborators{/collaborator}, hence stripping the end
+    project_users_url = project_users_url.split('{/')[0]
+
+    r = requests.get(project_users_url, headers={'Authorization': 'token ' + GITHUB_TOKEN})
+    r_data = r.json()
+
+    pr_appr_perm = [usr for usr in r_data if usr['login'] == pr_approver][0]['permissions']
+    if not pr_appr_perm['triage']:
+        raise SystemError("PR Approver Access is below TRIAGE level!")
+
+
+def setup_project(project_fullname):
+    print('Connecting to gitlab...')
+    GITLAB_URL = os.environ['GITLAB_URL']
+    GITLAB_TOKEN = os.environ['GITLAB_TOKEN']
+
+    gl = gitlab.Gitlab(url=GITLAB_URL, private_token=GITLAB_TOKEN)
+    gl.auth()
+
+    HDR_LEN = 8
+    gl_project_url = GITLAB_URL[: HDR_LEN] + GITLAB_TOKEN + ':' + GITLAB_TOKEN + '@' + GITLAB_URL[HDR_LEN :] + '/' + project_fullname + '.git'
+    
+    print(Git(".").clone(gl_project_url))
+    return gl
+
+
+# Merge PRs without Rebase (for new PRs)
+def sync_pr_with_merge(project_name, pr_num, pr_branch, project_html_url):
+    git = Git(project_name)
+    
+    GITHUB_REMOTE_NAME = 'github'
+    GITHUB_REMOTE_URL = project_html_url
+
+    HDR_LEN = 8
+    GITHUB_TOKEN = os.environ['GITHUB_TOKEN']
+    gh_remote = GITHUB_REMOTE_URL[: HDR_LEN] + GITHUB_TOKEN + ':' + GITHUB_TOKEN + '@' + GITHUB_REMOTE_URL[HDR_LEN :]
+    
+    print('Checking out to master branch...')
+    print(git.checkout('master'))
+
+    print('Adding the Github remote...')
+    print(git.remote('add', GITHUB_REMOTE_NAME, gh_remote))
+
+    print('Fetching the PR branch...')
+    print(git.fetch(GITHUB_REMOTE_NAME, 'pull/' + str(pr_num) + '/head'))
+
+    print('Checking out the PR branch...')
+    print(git.checkout('FETCH_HEAD', b=pr_branch))
+
+    print('Pushing to remote...')
+    print(git.push('--set-upstream', 'origin', pr_branch))
+
+
+# Merge PRs with Rebase approach (for old PRs)
+def sync_pr_with_rebase(project_name, pr_branch, pr_html_url, pr_rest_url):
+    git = Git(project_name)
+    repo = Repo(project_name)
+
+    #  Set the config parameters: Better be a espressif bot
+    repo.config_writer().set_value('user', 'name', os.environ['GIT_CONFIG_NAME']).release()
+    repo.config_writer().set_value('user', 'email', os.environ['GIT_CONFIG_EMAIL']).release()
+
+    # Download the patch for the given PR
+    pr_download_patch(pr_rest_url, project_name)
+
+    print('Checking out to master branch...')
+    print(git.checkout('master'))
+
+    print('Pulling the latest changes...')
+    print(git.pull('origin','master'))
+
+    print('Updating submodules...')
+    print(git.submodule('update', '--init', '--recursive'))
+
+    print('Checking out to new branch for contribution...')
+    print(git.checkout('HEAD', b=pr_branch))
+
+    print('Applying patch...')
+    print(git.execute(['git','am', 'diff.patch']))
```

Comment:
Alternative suggestion:
1. Invoke sync_pr_with_merge
2. rebase on top of latest master
3. continue with commit --amend

Reply:
Thank you for this suggestion.